python/ray/data/tests/test_arrow_serialization.py

```diff
@@ -0,0 +1,259 @@
+        (
+            pa.FixedSizeListArray.from_arrays(
+                pa.array([None] + [list(range(9)) + [None]] * 9), 10
+            ),
+            0.5,
+        ),
+        # Map array
+        (
+            pa.array(
+                [
+                    [(key, item) for key, item in zip("abcdefghij", range(10))]
+                    for _ in range(100)
+                ],
+                type=pa.map_(pa.string(), pa.int64()),
+            ),
+            0.5,
+        ),
+        # Struct array
+        (pa.array({"a": i} for i in range(100)), 0.5),
+        # Union array (sparse)
+        (
+            pa.UnionArray.from_sparse(
+                pa.array([0, 1] * 50, type=pa.int8()),
+                [pa.array(list(range(100))), pa.array([True, False] * 50)],
+            ),
+            0.5,
+        ),
+        # TODO(Clark): Support dense union arrays.
+        # # Union array (dense)
+        # (
+        #     pa.UnionArray.from_dense(
+        #         pa.array([0, 1] * 50, type=pa.int8()),
+        #         pa.array([
+        #             i if i % 2 == 0 else (i % 3) % 2
+        #             for i in range(100)
+        #         ], type=pa.int32()),
+        #         [pa.array(list(range(100))), pa.array([True, False])],
+        #     ),
+        #     0.5,
+        # ),
+        # Dictionary array
+        (
+            pa.DictionaryArray.from_arrays(
+                pa.array((list(range(9)) + [None]) * 10),
+                pa.array(["a", "b", "c", "d", "e", "f", "g", "h", "i"]),
+            ),
+            0.5,
+        ),
+        # Tensor extension array
+        (ArrowTensorArray.from_numpy(np.arange(100 * 4 * 4).reshape(100, 4, 4)), 0.5),
+        # Boolean tensor extension array
+        (
+            ArrowTensorArray.from_numpy(
+                np.array(
+                    [True, False, False, True, False, False, True, True] * 2 * 100
+                ).reshape(100, 4, 4)
+            ),
+            0.5,
+        ),
+        # Complex nested array
+        (
+            pa.UnionArray.from_sparse(
+                pa.array([0, 1] * 50, type=pa.int8()),
+                [
+                    pa.array(
+                        [
+                            {
+                                "a": i % 2 == 0,
+                                "b": i,
+                                "c": "bar",
+                            }
+                            for i in range(100)
+                        ]
+                    ),
+                    pa.array(
+                        [
+                            [(key, item) for key, item in zip("abcdefghij", range(10))]
+                            for _ in range(100)
+                        ],
+                        type=pa.map_(pa.string(), pa.int64()),
+                    ),
+                ],
+            ),
+            0.5,
+        ),
+    ],
+)
+def test_custom_arrow_array_serializer(ray_start_regular, arr, cap_mult):
```

Comment:
I like this test, please just make sure the custom array types are exhausted. Looks at least Date/Time are missing.

Reply:
I can add full coverage, although these _are_ the core array types, all others are slight variations of one of the above arrays. E.g., the date/time arrays are just another `NumericArray` specialization, so the only difference between those arrays and the int/float arrays will be the bit-width of the value type.